safenode/src/network/mod.rs

```diff
@@ -318,3 +326,109 @@ impl Network {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::SwarmDriver;
+    use crate::log::init_node_logging;
+    use eyre::{eyre, Result};
+    use libp2p::{
+        kad::kbucket::{Entry, InsertResult, KBucketsTable, Key, NodeStatus},
+        PeerId,
+    };
+    use rand::thread_rng;
+    use std::{
+        collections::{BTreeMap, HashMap},
+        fmt,
+        time::Duration,
+    };
+    use xor_name::XorName;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn closest() -> Result<()> {
+        let _ = init_node_logging(&None)?;
+        let mut networks_list = Vec::new();
+        let mut network_events_recievers = BTreeMap::new();
+        for _ in 1..25 {
```

Comment:
One small observation that I had is that when the node count is > 28, the node never really sends out the `Cmd::GetClosestPeer` and the whole test freezes. There are 0 log lines printed as well. I suspect it has something to do with all the nodes running inside the same process?

Reply:
Hm, that's unfortunate.